lib/streamlit/type_util.py

```diff
@@ -358,6 +358,16 @@ def is_pydeck(obj: object) -> TypeGuard[Deck]:
     return is_type(obj, "pydeck.bindings.deck.Deck")
 
 
+def is_iterable(obj: object) -> TypeGuard[Iterable[Any]]:
```

Comment:
Thought this was a neat abstraction, so moved the logic to `type_util`, so that it can be used in other places.